app/controllers/reviews_controller.rb

```diff
@@ -19,12 +16,12 @@ def create
   end
 
   def edit
-    @review = current_user.reviews.find(params[:id])
-    @buddy = Buddy.find(params[:buddy_id])
+    @review = Review.find(params[:id])
```

Comment:
Why did you undo the `current_user` scope? Without that anyone can access the edit page for a review.